utils/generator/transformers/structs_producer.py

```diff
@@ -0,0 +1,138 @@
+"""
+Structs transformation
+"""
+
+import logging
+import textwrap
+from collections import namedtuple, OrderedDict
+
+from model.param import Param
+from model.struct import Struct
+from transformers.common_producer import InterfaceProducerCommon
+
+
+class StructsProducer(InterfaceProducerCommon):
+    """
+    Structs transformation
+    """
+
+    def __init__(self, paths, enum_names, struct_names, mapping=None):
+        super(StructsProducer, self).__init__(
+            container_name='members',
+            enums_package=paths.enums_package,
+            structs_package=paths.structs_package,
+            enum_names=enum_names,
+            struct_names=struct_names,
+            package_name=paths.structs_package,
+            mapping=mapping['structs'] if mapping and 'structs' in mapping else {},
+            all_mapping=mapping if mapping else {})
+        self.logger = logging.getLogger('StructsProducer')
+        self.struct_class = paths.struct_class
+
+    def transform(self, item: Struct) -> dict:
+        """
+        Override
+        :param item: particular element from initial Model
+        :return: dictionary to be applied to jinja2 template
+        """
+        class_name = self.replace_sync(item.name[:1].upper() + item.name[1:])
+
+        imports = {'java.util.Hashtable'}
+        extends_class = self.struct_class
+
+        imports.add(extends_class)
+        extends_class = extends_class.rpartition('.')[-1]
+
+        params = OrderedDict()
+
+        for param in getattr(item, self.container_name).values():
+            param.name = self.replace_sync(param.name)
+            i, p = self.extract_param(param)
+            imports.update(i)
+            params[param.name] = p
+
+        render = OrderedDict()
+        render['class_name'] = class_name
+        render['extends_class'] = extends_class
+        render['package_name'] = self.package_name
+        render['imports'] = imports
```

Comment:
is not that duplicate of what is done in line 65? that applies to other producers as well.

Reply:
Removed duplication, thanks